src/redux/sagas/application-controller.ts

```diff
@@ -43,28 +42,32 @@ export default function* applicationControllerSaga(wallet: Wallet) {
         case GameActionType.CHOOSE_PLAY:
           newState = gameEngine.choosePlay(action.play);
           break;
-        case GameActionType.EVENT_RECEIVED:
-          newState = gameEngine.receiveEvent(action.event);
-          break;
         default:
-          // do nothing
+        // do nothing
       }
     }
-
     if (newState) {
-      if (newState.isReadyToSend) {
+      // Update that tehs tate has changed
+      yield put(GameAction.stateChanged(newState));
 
-        yield put(MessageAction.sendMessage(newState.opponentAddress, newState.move.toHex()))
-        yield put(GameAction.moveSent(newState.move));
-      }
-      // Fake sending a transaction for now
-      if (newState instanceof ReadyToDeploy && gameEngine){
-          gameEngine.transactionSent();
-          newState = gameEngine.state;
-      }
-      // TODO: Once we stop faking the transaction we won't need this null check
-      if (newState){
-        yield put(GameAction.stateChanged(newState));
+      if (gameEngine != null) {
+        if (newState.isReadyToSend) {
+          yield put(MessageAction.sendMessage(newState.opponentAddress, newState.move.toHex()));
+          yield put(GameAction.moveSent(newState.move));
+        }
+        if (newState.isReadyForFunding) {
+          yield put(WalletFundingAction.walletFundingRequest(wallet, newState.player));
+          const fundingAction: WalletFundingAction = yield take(
+            WalletFundingActionType.WALLETFUNDING_FUNDED,
+          );
```

Comment:
It looks to me like this is going to block until the wallet returns (right?). I'm wondering whether this is what we want to do, as (a) the game state won't be updated and (b) the application controller will become blocked during this time. Would it work to handle the incoming `WALLETFUNDING_FUNDED` action above in the regular `appliction-controller` processing loop instead?

Reply:
Good idea!